### PR TITLE
chore: refactor to put function name generator in constant

### PR DESCRIPTION
**Pull request by rishav-karanjit**
*Issue #, if available:*

*Description of changes:*
- AWS, local shape visitor helper and validation generator contains funcNameGenerator. So, brought it in a common constant file to avoid code duplication.
- ignoring java formating for this PR

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/awssdk/DafnyAwsSdkClientTypeConversionProtocol.java [hunk 1]
```diff
@@ -598,7 +600,7 @@ func Error_ToDafny(err error)($L.Error) {
             }
             """,
             DafnyNameResolver.dafnyTypesNamespace(serviceShape),
-            writer.consumer(w -> {
+              writer.consumer(w -> {
```
Comment (rishav-karanjit) [inline codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/awssdk/DafnyAwsSdkClientTypeConversionProtocol.java]: Not sure how this came in diff but make format_java will fix this. So, Ignoring this for now.

File: codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/awssdk/DafnyAwsSdkClientTypeConversionProtocol.java [hunk 2]
```diff
@@ -807,7 +809,7 @@ private void generateSerializerFunctions(final GenerationContext context, final
                   return $L
               }
               """,
-            ShapeVisitorHelper.funcNameGenerator(
```
Comment (ShubhamChaturvedi7) [inline codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/awssdk/DafnyAwsSdkClientTypeConversionProtocol.java:826]: nit: I'd prefer fully-qualified-name / FQN over static imports. Improves readability / reasoning around where this function belongs.

File: codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/awssdk/DafnyAwsSdkClientTypeConversionProtocol.java (hunk 1 above)
Comment (ShubhamChaturvedi7) [reply codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/awssdk/DafnyAwsSdkClientTypeConversionProtocol.java]: Personally I'm not a fan of our java formatter - it keeps line too short and I think uses 2 white-spaces for indentation.

File: codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/utils/Constants.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    public static String funcNameGenerator(
+    final MemberShape memberShape,
+    final String suffix
+    ) {
+        return memberShape
+        .getId()
+        .toString()
+        .replaceAll("[.$#]", "_")
+        .concat("_")
+        .concat(suffix);
```
Comment (ShubhamChaturvedi7) [inline codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/utils/Constants.java]: Why is this different than one in the `ValidationGenerator` ?

Comment (rishav-karanjit) [reply codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/utils/Constants.java]: I thought the name in ValidationGenerator won't collide and will look good but the name collided (Caught in CI in [8738b87](https://github.com/smithy-lang/smithy-dafny/pull/641/commits/8738b875b4e1d6678bbcf7ee2c5d4736a050d2d9)) So, I made ValidationGenerator to use same function name.

File: codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/awssdk/DafnyAwsSdkClientTypeConversionProtocol.java (hunk 2 above)
Comment (rishav-karanjit) [reply codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/awssdk/DafnyAwsSdkClientTypeConversionProtocol.java:826]: Done

**Review by ShubhamChaturvedi7 — APPROVED** [review]
Follow this up with the formatting fix.

**Review by ShubhamChaturvedi7 — APPROVED** [review]

**Pull request merged**